api/db/populate_users.js

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env node
+
+// TODO: This will all be in the general js to abide by DRY principal
+const userArgs = process.argv.slice(2);
+
+if (!userArgs[0].startsWith('mongodb')) {
+  console.log(
+    'Error: you need to specify a valid MongoDB URL as the first argument'
+  );
+
+  return;
+}
+
+const User = require('../models/User');
+const Interest = require('../models/Interest')
+const Token = require('../models/Token')
+const crypto = require('crypto');
+
+const async = require('async');
+
+const mongoose = require('mongoose');
+const mongoDB = userArgs[0];
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = global.Promise;
+const db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+const { uniqueNamesGenerator, names } = require('unique-names-generator');
+const countries = require('country-region-data');
+const locales = ['en', 'es'];
+const users = [];
+
+const createUser = async (
+  name, email, emailConfirmed, emailConfirmedAt,
+    dateOfBirth, country, region,
+    preferredLocale, interests, admin,
+    callback
+) => {
+  const user = new User({
+    name, email, emailConfirmed, emailConfirmedAt,
+    dateOfBirth, country, region,
+    preferredLocale, interests, admin
+  });
+  user.setPassword('Password1');
+  await user.save();
+  const token = createTokenForUser(user);
+  await token.save();
+
+  console.log(`User: ${user}`);
+  users.push(user);
+  callback(null, user);
+}
+
+function randomDate(start, end) {
+  return new Date(+start + Math.random() * (end - start));
+}
+
+createTokenForUser = (user) => {
+  return new Token({
+    user,
+    type: 'Email',
+    token: crypto.randomBytes(16).toString('hex')
+  });
+}
+
+const generateUsersArray = async () => {
+  let interests = await Interest.distinct("_id");
+  const config = {
+    dictionaries: [names]
+  }
+
+  let arr = []
+  for (let i = 0; i < 100; i++) {
+    const name = `${uniqueNamesGenerator(config)} ${uniqueNamesGenerator(config)}`;
+    const email = `${name.replace(/ /g, '_')}@test-tisn-email.com`.toLowerCase();
+    const country = countries[Math.floor(Math.random() * countries.length)];
+    const region = country.regions[Math.floor(Math.random() * country.regions.length)];
+    const enrolledDate = randomDate(new Date(2019, 01, 01), new Date());
+    const birthDate = randomDate(new Date(1950, 01, 01), new Date(2007, 01, 01));
+    interests = interests.sort(() => 0.5 - Math.random());
+    const selectedInterests = interests.slice(0, Math.floor(Math.random() * 10));
+
+
+    arr.push(
+      (seriesCallback) => {
+        createUser(
+          name,
+          email,
+          true,
+          enrolledDate,
+          birthDate,
+          country.countryShortCode,
+          region.shortCode || 'N/A',
```

Comment:
That is correct, there are some countries that do not have any regions. 

I was going to bring that one up as a bug along with another regarding emails.

Reply:
Sounds good! Let's take care of that on its own issue then 😊

Reply:
Are you opening this issue, or should I? 🙂

Reply:
If you don't mind doing that. You may have better insight as to where else in the code it may be causing issues.

Reply:
Ok, will do! 🙂

Reply:
Oh, it seems this won't be an issue in the end, as countries with no regions seem to have this structure:

```javascript
{
  countryName: countryName
  countryShortCode: countryShortCode
  regions: [{
    name: countryName
    shortCode: countryShortCode
  }]
}
```

So I think we can just get rid of the ` || 'N/A'` part! 🙂

Reply:
Sorry, I missed this. That may be the case, but when I left the field null it would fail to create the document. It is likely because the model is requiring the region to have something.

```
const UserSchema = new Schema(
  {
    name: { type: String, required: true },
    email: { type: String, required: true },
    emailConfirmed: { type: Boolean, required: true, default: false },
    emailConfirmedAt: { type: Date },
    salt: { type: String, required: true },
    hash: { type: String, required: true },
    dateOfBirth: { type: Date, required: true },
    country: { type: String, required: true },
    region: { type: String, required: true },
    preferredLocale: { type: String, required: true, enum: ['en', 'es'] },
    avatar: { type: String },
    interests: [{ type: Schema.Types.ObjectId, ref: 'Interest' }],
    admin: { type: Boolean, required: true, default: false },
  },
  { timestamps: true }
);

Reply:
Ah, I see the problem now -- the problem is not countries without regions, but regions without short codes instead. How does using `region: region.shortCode || region.name` sound?

Reply:
That works for me. Good eye! I'll make that change now.

Reply:
Perfect! I'll open an issue regarding this now, as it's indeed a problem in other parts of the app too.